src/components/SeatContest.tsx

```diff
@@ -63,59 +66,156 @@ const Choice = styled('label')<{ isSelected: boolean }>`
   & > div:first-of-type {
     padding-left: 4rem;
   }
-  // & .candidate-party {
-  //   text-align: right;
-  // }
+  & .write-in-candidate-name {
+    grid-column: span 2;
+  }
 `
 const ChoiceInput = styled.input.attrs({
   type: 'checkbox',
 })`
   margin-right: 0.5rem;
 `
 
+const WriteInCandidateFieldSet = styled.fieldset`
+  margin: 1rem 0;
+  border: none;
+  padding: 0;
+`
+
+const WriteInCandidateInput = styled.input.attrs({
+  readOnly: true,
+  type: 'text',
+})`
+  outline: none;
+  width: 100%;
+`
+
 interface Props {
   contest: Contest
   vote: Vote
   updateVote: UpdateVoteFunction
 }
 
 interface State {
-  attemptedCandidateSelection: string
+  attemptedVoteCandidateName: string
+  writeInCandateModalIsOpen: boolean
+  writeInCandidateName: string
+  layoutName: string
 }
 
 const initialState = {
-  attemptedCandidateSelection: '',
+  attemptedVoteCandidateName: '',
+  layoutName: 'default',
+  writeInCandateModalIsOpen: false,
+  writeInCandidateName: '',
 }
 
 class SeatContest extends React.Component<Props, State> {
-  public state: State = initialState
+  private keyboard: React.RefObject<Keyboard>
+  constructor(props: Props) {
+    super(props)
+    let writeInCandidateName = ''
```

Comment:
are we using empty string as the null write-in candidate name? That freaks me out a little bit. Should we not simply make this nullable?

Reply:
Hmmm… Typescript is config'd to never use `null` only `undefined`. It's simpler Typescript code if types to not change.

Can you provide repro steps to show how things could go wrong?

Perhaps we `trim()` the string upon submission to catch any string containing spaces? 

Otherwise, I don't see how an empty string is any worse than `undefined‘.

Reply:
Cool. I've added some checks. Pushing new code shortly.